src/app/clusters/thread-border-router-management-server/thread-br-delegate.h

```diff
@@ -0,0 +1,76 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-enums.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <inet/IPAddress.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/Optional.h>
+#include <lib/support/ScopedBuffer.h>
+#include <lib/support/Span.h>
+#include <lib/support/ThreadOperationalDataset.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ThreadBorderRouterManagement {
+
+constexpr size_t kBorderRouterNameMaxLength = 63;
+constexpr size_t kBorderAgentIdLength       = 16;
+
+class Delegate
+{
+public:
+    Delegate()          = default;
+    virtual ~Delegate() = default;
+
+    class ActivateDatasetCallback
+    {
+    public:
+        virtual void OnActivateDatasetComplete(CHIP_ERROR error) = 0;
+    };
+
+    virtual CHIP_ERROR Init() = 0;
+
+    virtual CHIP_ERROR GetPanChangeSupported(bool & panChangeSupported) = 0;
+
+    virtual CHIP_ERROR GetBorderRouterName(MutableCharSpan & borderRouterName) = 0;
+
+    virtual CHIP_ERROR GetBorderAgentId(MutableByteSpan & borderAgentId) = 0;
+
+    virtual CHIP_ERROR GetThreadVersion(uint16_t & threadVersion) = 0;
+
+    virtual CHIP_ERROR GetInterfaceEnabled(bool & interfaceEnabled) = 0;
+
+    virtual CHIP_ERROR GetActiveDataset(chip::Thread::OperationalDataset & activeDataset) = 0;
+
+    virtual CHIP_ERROR GetPendingDataset(chip::Thread::OperationalDataset & pendingDataset) = 0;
+
+    virtual CHIP_ERROR SetActiveDataset(const chip::Thread::OperationalDataset & activeDataset,
+                                        ActivateDatasetCallback * callback) = 0;
+
+    virtual CHIP_ERROR RevertActiveDataset() = 0;
```

Comment:
This needs to clearly document some behavior expectations (e.g. after a call to RevertActiveDataset there must be no calls to the ActivateDatasetCallback previously passed to SetActiveDataset or so, yes?).

Reply:
Added the comments